File: DotNetGotchas/VBDN/CopyingObjects/Copy/Brain.vb
'Brain.vb
Public Class Brain
    Public Sub New()

    End Sub

    Public Sub New(ByVal another As Brain)
        ' Code to properly copy Brain can go here
    End Sub

    Public Overrides Function ToString() As String
        Return Me.GetType().Name & ":" & GetHashCode()
    End Function

End Class

File: DotNetGotchas/CSharp/CopyingObjects/Modified2/Copy/Brain.cs
//Brain.cs
using System;

namespace Copy
{
	public class Brain
	{
		public Brain() {}

		public Brain(Brain another) 
		{
			//Code to properly copy Brain can go here
		}

		public override string ToString()
		{
			return GetType().Name + ":" + GetHashCode();
		}

	}
}
